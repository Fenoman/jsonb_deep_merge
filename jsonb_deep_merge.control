#jsonb_deep_merge extension
comment = 'jsonb deep merge'
default_version = '0.0.1'

