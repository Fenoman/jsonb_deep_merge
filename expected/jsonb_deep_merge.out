CREATE EXTENSION jsonb_deep_merge;
select jsonb_deep_merge('{}'::jsonb, '{}'::jsonb);
 jsonb_deep_merge 
------------------
 {}
(1 row)

select jsonb_deep_merge('{"a": 1}'::jsonb, '{}'::jsonb);
 jsonb_deep_merge 
------------------
 {"a": 1}
(1 row)

select jsonb_deep_merge('{}'::jsonb, '{"a": 1}'::jsonb);
 jsonb_deep_merge 
------------------
 {"a": 1}
(1 row)

select jsonb_deep_merge('{"b": 2}'::jsonb, '{"a": 1}'::jsonb);
 jsonb_deep_merge 
------------------
 {"a": 1, "b": 2}
(1 row)

select jsonb_deep_merge('{"b": 2, "a": 1}'::jsonb, '{"a": "3"}'::jsonb);
  jsonb_deep_merge  
--------------------
 {"a": "3", "b": 2}
(1 row)

select jsonb_deep_merge('{"b": {"c": 2}, "a": 1}'::jsonb, '{"a": {"c": 1}}'::jsonb);
        jsonb_deep_merge        
--------------------------------
 {"a": {"c": 1}, "b": {"c": 2}}
(1 row)

select jsonb_deep_merge('{"b": {"c": 2}}'::jsonb, '{"a": 1}'::jsonb);
    jsonb_deep_merge     
-------------------------
 {"a": 1, "b": {"c": 2}}
(1 row)

select jsonb_deep_merge('{"b": {"c": 2}}'::jsonb, '{"b": {"a": 1}}'::jsonb);
    jsonb_deep_merge     
-------------------------
 {"b": {"a": 1, "c": 2}}
(1 row)

select jsonb_deep_merge('{"a": 1, "b": {"c": 3}}'::jsonb, '{"b": {"d": 4}}'::jsonb);
        jsonb_deep_merge         
---------------------------------
 {"a": 1, "b": {"c": 3, "d": 4}}
(1 row)

select jsonb_deep_merge('{"a": {"b": 1, "c": 2}}'::jsonb, '{"a": 23}'::jsonb);
 jsonb_deep_merge 
------------------
 {"a": 23}
(1 row)

select jsonb_deep_merge('{"a": 23}'::jsonb, '{"a": {"b": 1, "c": 2}}'::jsonb);
    jsonb_deep_merge     
-------------------------
 {"a": {"b": 1, "c": 2}}
(1 row)

select jsonb_deep_merge('{"a": {"c": 23}}'::jsonb, '{"a": {"b": 1, "c": {"d": 2}}}'::jsonb);
        jsonb_deep_merge        
--------------------------------
 {"a": {"b": 1, "c": {"d": 2}}}
(1 row)

select jsonb_deep_merge('{"a": {"b": 1, "c": {"d": 2}}}'::jsonb, '{"a": {"c": 23}}'::jsonb);
     jsonb_deep_merge     
--------------------------
 {"a": {"b": 1, "c": 23}}
(1 row)

select jsonb_deep_merge('{"a": {"b": 1}}', '{"a": {"b": false}}');
 jsonb_deep_merge 
------------------
 {}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11}'::jsonb, '{"g" : {"b" : null, "f" : false}}'::jsonb);
         jsonb_deep_merge         
----------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11}
(1 row)

select jsonb_deep_merge('{"g" : {"b" : null, "f" : false}}':: jsonb, '{"a": {"b" : 2, "f" : 2}, "c" : 11}'::jsonb);
                  jsonb_deep_merge                   
-----------------------------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11, "g": {"f": false}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11}'::jsonb, '{"a" : {"b" : null, "f" : false}}'::jsonb);
 jsonb_deep_merge 
------------------
 {"c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "f" : false}}'::jsonb, '{"a" : {"b" : 2, "f" : 2}, "c" : 11}'::jsonb);
         jsonb_deep_merge         
----------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "f" : false}}'::jsonb);
   jsonb_deep_merge    
-----------------------
 {"c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "f" : false}}'::jsonb, '{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb);
         jsonb_deep_merge         
----------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false}}'::jsonb);
           jsonb_deep_merge           
--------------------------------------
 {"a": {"f": 2}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false}}'::jsonb, '{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb);
               jsonb_deep_merge               
----------------------------------------------
 {"a": {"b": 2, "d": false, "f": 2}, "c": 11}
(1 row)

select jsonb_deep_merge('{"b" : 2, "k" : {"k1" : 1, "k2" : 2}}'::jsonb, '{"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}'::jsonb);
 jsonb_deep_merge 
------------------
 {}
(1 row)

select jsonb_deep_merge('{"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}'::jsonb, '{"b" : 2, "k" : {"k1" : 1, "k2" : 2}}'::jsonb);
               jsonb_deep_merge                
-----------------------------------------------
 {"b": 2, "d": false, "k": {"k1": 1, "k2": 2}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, '{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb);
 jsonb_deep_merge 
------------------
 {"a": {"k": {}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb);
                   jsonb_deep_merge                   
------------------------------------------------------
 {"a": {"b": 2, "d": false, "k": {"k1": 1, "k2": 2}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, '{"a" : {"b" : null, "d" : 2, "k" : {"k1" : null, "k2" : false}}}'::jsonb);
     jsonb_deep_merge     
--------------------------
 {"a": {"d": 2, "k": {}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : 2, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb);
                 jsonb_deep_merge                 
--------------------------------------------------
 {"a": {"b": 2, "d": 2, "k": {"k1": 1, "k2": 2}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, '{"z" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb);
                     jsonb_deep_merge                     
----------------------------------------------------------
 {"a": {"b": 2, "k": {"k1": 1, "k2": 2}}, "z": {"k": {}}}
(1 row)

select jsonb_deep_merge('{"z" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb);
                                jsonb_deep_merge                                 
---------------------------------------------------------------------------------
 {"a": {"b": 2, "k": {"k1": 1, "k2": 2}}, "z": {"d": false, "k": {"k2": false}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "k3" : 2}}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb);
               jsonb_deep_merge               
----------------------------------------------
 {"a": {"k": {"k3": 2}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "k3" : 2}}, "c" : 11, "z" : false}'::jsonb);
                            jsonb_deep_merge                            
------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "k": {"k1": 1, "k2": 2, "k3": 2}}, "c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false, "n" : {"m" : {}}, "p" : {"k1" : false, "k2" : null}}}'::jsonb);
                            jsonb_deep_merge                            
------------------------------------------------------------------------
 {"a": {"f": 2, "k": {"k1": 1, "k2": 2}, "n": {}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "n" : {"m" : {}}, "p" : {"k1" : false, "k2" : null}}}'::jsonb, '{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb);
                                          jsonb_deep_merge                                          
----------------------------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "f": 2, "k": {"k1": 1, "k2": 2}, "n": {}}, "c": 11, "p": {"k1": false}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false, "p" : {"k1" : 5, "k2" : null, "k3": {"kz1" : 1, "kz2" : null}}}}'::jsonb);
                                        jsonb_deep_merge                                         
-------------------------------------------------------------------------------------------------
 {"a": {"f": 2, "k": {"k1": 1, "k2": 2}, "p": {"k1": 5, "k3": {"kz1": 1}}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "p" : {"k1" : 5, "k2" : null, "k3": {"kz1" : 1, "kz2" : null}}}}'::jsonb, '{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb);
                                            jsonb_deep_merge                                             
---------------------------------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "f": 2, "k": {"k1": 1, "k2": 2}, "p": {"k1": 5, "k3": {"kz1": 1}}}, "c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "l" : {"l1" : 1, "l2" : 2}}}}'::jsonb, '{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false, "l" : {"l1" : null, "l2" : false}}}}'::jsonb);
    jsonb_deep_merge     
-------------------------
 {"a": {"k": {"l": {}}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false, "l" : {"l1" : null, "l2" : false}}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "l" : {"l1" : 1, "l2" : 2}}}}'::jsonb);
                               jsonb_deep_merge                                
-------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "k": {"l": {"l1": 1, "l2": 2}, "k1": 1, "k2": 2}}}
(1 row)

select jsonb_deep_merge('{}'::jsonb, '{}'::jsonb, false);
 jsonb_deep_merge 
------------------
 {}
(1 row)

select jsonb_deep_merge('{"a": 1}'::jsonb, '{}'::jsonb, false);
 jsonb_deep_merge 
------------------
 {"a": 1}
(1 row)

select jsonb_deep_merge('{}'::jsonb, '{"a": 1}'::jsonb, false);
 jsonb_deep_merge 
------------------
 {"a": 1}
(1 row)

select jsonb_deep_merge('{"b": 2}'::jsonb, '{"a": 1}'::jsonb, false);
 jsonb_deep_merge 
------------------
 {"a": 1, "b": 2}
(1 row)

select jsonb_deep_merge('{"b": 2, "a": 1}'::jsonb, '{"a": "3"}'::jsonb, false);
  jsonb_deep_merge  
--------------------
 {"a": "3", "b": 2}
(1 row)

select jsonb_deep_merge('{"b": {"c": 2}, "a": 1}'::jsonb, '{"a": {"c": 1}}'::jsonb, false);
        jsonb_deep_merge        
--------------------------------
 {"a": {"c": 1}, "b": {"c": 2}}
(1 row)

select jsonb_deep_merge('{"b": {"c": 2}}'::jsonb, '{"a": 1}'::jsonb, false);
    jsonb_deep_merge     
-------------------------
 {"a": 1, "b": {"c": 2}}
(1 row)

select jsonb_deep_merge('{"b": {"c": 2}}'::jsonb, '{"b": {"a": 1}}'::jsonb, false);
    jsonb_deep_merge     
-------------------------
 {"b": {"a": 1, "c": 2}}
(1 row)

select jsonb_deep_merge('{"a": 1, "b": {"c": 3}}'::jsonb, '{"b": {"d": 4}}'::jsonb, false);
        jsonb_deep_merge         
---------------------------------
 {"a": 1, "b": {"c": 3, "d": 4}}
(1 row)

select jsonb_deep_merge('{"a": {"b": 1, "c": 2}}'::jsonb, '{"a": 23}'::jsonb, false);
 jsonb_deep_merge 
------------------
 {"a": 23}
(1 row)

select jsonb_deep_merge('{"a": 23}'::jsonb, '{"a": {"b": 1, "c": 2}}'::jsonb, false);
    jsonb_deep_merge     
-------------------------
 {"a": {"b": 1, "c": 2}}
(1 row)

select jsonb_deep_merge('{"a": {"c": 23}}'::jsonb, '{"a": {"b": 1, "c": {"d": 2}}}'::jsonb, false);
        jsonb_deep_merge        
--------------------------------
 {"a": {"b": 1, "c": {"d": 2}}}
(1 row)

select jsonb_deep_merge('{"a": {"b": 1, "c": {"d": 2}}}'::jsonb, '{"a": {"c": 23}}'::jsonb, false);
     jsonb_deep_merge     
--------------------------
 {"a": {"b": 1, "c": 23}}
(1 row)

select jsonb_deep_merge('{"a": {"b": 1}}', '{"a": {"b": false}}', false);
  jsonb_deep_merge   
---------------------
 {"a": {"b": false}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11}'::jsonb, '{"g" : {"b" : null, "f" : false}}'::jsonb, false);
                  jsonb_deep_merge                   
-----------------------------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11, "g": {"f": false}}
(1 row)

select jsonb_deep_merge('{"g" : {"b" : null, "f" : false}}':: jsonb, '{"a": {"b" : 2, "f" : 2}, "c" : 11}'::jsonb, false);
                  jsonb_deep_merge                   
-----------------------------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11, "g": {"f": false}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11}'::jsonb, '{"a" : {"b" : null, "f" : false}}'::jsonb, false);
       jsonb_deep_merge       
------------------------------
 {"a": {"f": false}, "c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "f" : false}}'::jsonb, '{"a" : {"b" : 2, "f" : 2}, "c" : 11}'::jsonb, false);
         jsonb_deep_merge         
----------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "f" : false}}'::jsonb, false);
             jsonb_deep_merge             
------------------------------------------
 {"a": {"f": false}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "f" : false}}'::jsonb, '{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb, false);
               jsonb_deep_merge               
----------------------------------------------
 {"a": {"b": 2, "f": 2}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false}}'::jsonb, false);
                 jsonb_deep_merge                 
--------------------------------------------------
 {"a": {"d": false, "f": 2}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false}}'::jsonb, '{"a" : {"b" : 2, "f" : 2}, "c" : 11, "z" : false}'::jsonb, false);
                     jsonb_deep_merge                     
----------------------------------------------------------
 {"a": {"b": 2, "d": false, "f": 2}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"b" : 2, "k" : {"k1" : 1, "k2" : 2}}'::jsonb, '{"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}'::jsonb, false);
         jsonb_deep_merge         
----------------------------------
 {"d": false, "k": {"k2": false}}
(1 row)

select jsonb_deep_merge('{"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}'::jsonb, '{"b" : 2, "k" : {"k1" : 1, "k2" : 2}}'::jsonb, false);
               jsonb_deep_merge                
-----------------------------------------------
 {"b": 2, "d": false, "k": {"k1": 1, "k2": 2}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, '{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, false);
            jsonb_deep_merge             
-----------------------------------------
 {"a": {"d": false, "k": {"k2": false}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, false);
                   jsonb_deep_merge                   
------------------------------------------------------
 {"a": {"b": 2, "d": false, "k": {"k1": 1, "k2": 2}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, '{"a" : {"b" : null, "d" : 2, "k" : {"k1" : null, "k2" : false}}}'::jsonb, false);
          jsonb_deep_merge           
-------------------------------------
 {"a": {"d": 2, "k": {"k2": false}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : 2, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, false);
                 jsonb_deep_merge                 
--------------------------------------------------
 {"a": {"b": 2, "d": 2, "k": {"k1": 1, "k2": 2}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, '{"z" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, false);
                                jsonb_deep_merge                                 
---------------------------------------------------------------------------------
 {"a": {"b": 2, "k": {"k1": 1, "k2": 2}}, "z": {"d": false, "k": {"k2": false}}}
(1 row)

select jsonb_deep_merge('{"z" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2}}}'::jsonb, false);
                                jsonb_deep_merge                                 
---------------------------------------------------------------------------------
 {"a": {"b": 2, "k": {"k1": 1, "k2": 2}}, "z": {"d": false, "k": {"k2": false}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "k3" : 2}}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, false);
                           jsonb_deep_merge                            
-----------------------------------------------------------------------
 {"a": {"d": false, "k": {"k2": false, "k3": 2}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "k3" : 2}}, "c" : 11, "z" : false}'::jsonb, false);
                                  jsonb_deep_merge                                  
------------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "k": {"k1": 1, "k2": 2, "k3": 2}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false, "n" : {"m" : {}}, "p" : {"k1" : false, "k2" : null}}}'::jsonb, false);
                                            jsonb_deep_merge                                            
--------------------------------------------------------------------------------------------------------
 {"a": {"d": false, "f": 2, "k": {"k1": 1, "k2": 2}, "n": {}}, "c": 11, "p": {"k1": false}, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "n" : {"m" : {}}, "p" : {"k1" : false, "k2" : null}}}'::jsonb, '{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb, false);
                                                jsonb_deep_merge                                                
----------------------------------------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "f": 2, "k": {"k1": 1, "k2": 2}, "n": {}}, "c": 11, "p": {"k1": false}, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb, '{"a" : {"b" : null, "d" : false, "p" : {"k1" : 5, "k2" : null, "k3": {"kz1" : 1, "kz2" : null}}}}'::jsonb, false);
                                              jsonb_deep_merge                                               
-------------------------------------------------------------------------------------------------------------
 {"a": {"d": false, "f": 2, "k": {"k1": 1, "k2": 2}, "p": {"k1": 5, "k3": {"kz1": 1}}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "p" : {"k1" : 5, "k2" : null, "k3": {"kz1" : 1, "kz2" : null}}}}'::jsonb, '{"a" : {"b" : 2, "f" : 2, "k" : {"k1" : 1, "k2" : 2}}, "c" : 11, "z" : false}'::jsonb, false);
                                                  jsonb_deep_merge                                                   
---------------------------------------------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "f": 2, "k": {"k1": 1, "k2": 2}, "p": {"k1": 5, "k3": {"kz1": 1}}}, "c": 11, "z": false}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "l" : {"l1" : 1, "l2" : 2}}}}'::jsonb, '{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false, "l" : {"l1" : null, "l2" : false}}}}'::jsonb, false);
                      jsonb_deep_merge                       
-------------------------------------------------------------
 {"a": {"d": false, "k": {"l": {"l2": false}, "k2": false}}}
(1 row)

select jsonb_deep_merge('{"a" : {"b" : null, "d" : false, "k" : {"k1" : null, "k2" : false, "l" : {"l1" : null, "l2" : false}}}}'::jsonb, '{"a" : {"b" : 2, "k" : {"k1" : 1, "k2" : 2, "l" : {"l1" : 1, "l2" : 2}}}}'::jsonb, false);
                               jsonb_deep_merge                                
-------------------------------------------------------------------------------
 {"a": {"b": 2, "d": false, "k": {"l": {"l1": 1, "l2": 2}, "k1": 1, "k2": 2}}}
(1 row)

